alfalfa_worker/dispatcher.py

```diff
@@ -0,0 +1,139 @@
+#  derived from this software without specific prior written permission from the respective party.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+#  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+#  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+#  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+#  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+#  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+########################################################################################################################
+
+import traceback
+import json
+
+from pathlib import Path
+
+# Currently this is a child of WorkerJobBase, but mostly for the
+# alfalfa connections. WorkerJobBase could/should be updated to inherit
+# from a new class that just handles the alfalfa connections, then
+# the WorkerJobBase and Dispatcher can both inherit from the new class.
+from alfalfa_worker.lib.alfalfa_connections_base import AlfalfaConnectionsBase
+
+# Workers that are defined in this dispatcher
+from alfalfa_worker.worker_openstudio.worker import WorkerOpenStudio
+from alfalfa_worker.worker_fmu.worker import WorkerFmu
+from alfalfa_worker.lib.logger_mixins import DispatcherLoggerMixin
+
+
+class Dispatcher(DispatcherLoggerMixin, AlfalfaConnectionsBase):
+    """Class to pop data off a queue and determine to where the work needs
+    to go. Alfalfa works off a single queue and the work is identified
+    based on the payload that is in the queue (model_name).
+
+    This class is currently designed to do the work on the worker that it is
+    attached to. That is, the Dispatcher is not designed to pop work off the
+    queue and submit the job to another worker with the requisite files.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.logger.info(f"Job queue url is {self.sqs_queue}")
+
+        # create classes for the workers that this dispatcher supports
+        self.worker_openstudio_class = WorkerOpenStudio()
+        self.worker_fmu_class = WorkerFmu()
+
+    def determine_worker_class(self, test_str):
+        """This is the only place where we should decide
+        if the work is an OSW or an FMU
+        """
+        worker_class = None
+        test_str = test_str.lower().replace('.', '')
+        if test_str in ['osw', 'osm', 'zip']:
```

Comment:
OSM is still needed because after a site is added, the run_sim passes an OSM as the job type, which hits this method.